contracts/InverseVester.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.3;
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/IInv.sol";
+
+/**
+ * @title Inverse token vesting contract
+ * @author Inverse Finance
+ * @notice Base contract for vesting agreement on INV tokens
+ *         Reverse vesting means tokens are owned but the recipient
+ *         and can be used for voting.
+ * @dev    Vesting calculation is linear
+ */
+contract InverseVester is Ownable {
+    using SafeERC20 for IInv;
+
+    uint256 public constant DAY = 1 days;
+
+    // @dev Total amount to be vested
+    uint256 public vestingAmount;
+
+    // @dev Timestamp for the start of this vesting agreement
+    uint256 public vestingBegin;
```

Comment:
Can save some gas by lowering the bits of `vestingBegin` and `vestingEnd`, and combining `vestingDurationInDays` and `reverseVesting` into a struct, e.g.:
```solidity
struct VestingData {
    uint64 vestingBegin;
    uint64 vestingEnd;
    uint16 vestinDurationInDays;
    bool reverseVesting;
}
```
The `uint64` are just examples. As long as the total bits fit within a multiple of 256, they will take up the same 32 byte storage slot on-chain.

Here's a quick benchmark run I did:
```solidity
pragma solidity ^0.8.3;

contract Foo {
    uint256 public a;
    uint256 public b;
    
    struct Bar {
        uint128 a;
        uint128 b;
    }
    
    Bar public c;
    
    constructor() {
        a = 1;
        b = 2;
        c = Bar({
            a: 1,
            b: 2
        });
    }
    
    function x(uint256 _a, uint256 _b) external { // 32,307 gas
        a = _a;
        b = _b;
    }
    
    function y(uint128 _a, uint128 _b) external { // 29,821 gas
        c.a = _a;
        c.b = _b;
    }
    
    function z(uint128 _a, uint128 _b) external { // 30,008 gas
        c = Bar({
            a: _a,
            b: _b
        });
    }
}
```

Probably best to keep `lastClaimTimestamp` out of it due to it being the only state attribute that is mutable.

**BUT...**
The gas savings of a struct probably won't way up to the hassle of changing the contract and tests, though. :')
In which case, the easier change is to make `vestingBegin`, `vestingDurationInDays` and `reverseVesting` immutable.

Reply:
Sure can be optimised a bit, I supposed gas efficiency isn't a big factor given frequency of ops and amounts involved